datafusion/sql/src/expr/mod.rs

```diff
@@ -170,6 +170,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         schema: &DFSchema,
         planner_context: &mut PlannerContext,
     ) -> Result<Expr> {
+        let sql_not_moved = sql.clone();
```

Comment:
This isn't good. We may want something like https://github.com/VitWW/rfcs/blob/partial_types3/text/0000-partial_types.md

to implement partial move...

Reply:
Using references is also bad. The code moves fields from sqlparser::ast::Expr everywhere.